### PR TITLE
docs: add a guide to help those familiar with testing library

### LINKED ISSUE #13317: [Question] Is there any possibility for using TestingLibrary as a selector library?
We used [TestingLibrary](https://testing-library.com) as a nice solution for querying our React application structure in Jest and Cypress. In a new project I'll switch to Playwright, and I am looking for ways of integrating this very ARIA-focused way of accessing elements.

### PR DESCRIPTION
**Pull request by dgozman**
Closes #13317.

### CODE REVIEW

File: docs/src/testing-library-js.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+// Playwright
+await expect(page.locator('text=the lion king')).toBeVisible()
+await expect(page.locator('text=the mummy')).toBeHidden()
+```
+
+When you cannot find a suitable assertion, use [`expect.poll`](./test-assertions#polling) instead.
+
+```js
+expect.poll(async () => {
```
Comment (mxschmitt) [inline docs/src/testing-library-js.md:112]:
```suggestion
await expect.poll(async () => {
```
I think this is required.

File: docs/src/testing-library-js.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+const messages = page.locator('id=messages')
+const helloMessage = messages.locator('text=hello')
+```
+
+## Playwright Test Super Powers
+
+Once you're on Playwright Test, you get a lot!
+
+- Full zero-configuration TypeScript support
+- Run tests across **all web engines** (Chrome, Firefox, Safari) on **any popular operating system** (Windows, MacOS, Ubuntu)
```
Comment (mxschmitt) [inline docs/src/testing-library-js.md:137]:
```suggestion
- Run tests across **all web engines** (Chrome, Firefox, Safari) on **any popular operating system** (Windows, macOS, Ubuntu)
```

**Review by mxschmitt — APPROVED** [review]

File: docs/src/testing-library-js.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
+---
+id: testing-library
+title: "Migrating from Testing Library"
+---
+
+<!-- TOC -->
+
+## Migration Principes
+
+If you use DOM Testing Library in the browser (for example, you bundle tests with webpack), follow this guide for straightforward migration to Playwright Test.
```
Comment (rwoll) [inline docs/src/testing-library-js.md:10]:
```suggestion
If you use [DOM Testing Library](https://testing-library.com/docs/dom-testing-library/intro/) in the browser (for example, you bundle tests with webpack), follow this guide for straightforward migration to Playwright Test.
```
```suggestion
If you use DOM Testing Library in the browser (for example, you bundle tests with webpack), follow this guide for straightforward migration to Playwright Test.
```

**Review by rwoll — APPROVED** [review]

File: docs/src/testing-library-js.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+const helloMessage = within(messages).getByText('hello')
+
+// Playwright
+const messages = page.locator('id=messages')
+const helloMessage = messages.locator('text=hello')
+```
+
+## Playwright Test Super Powers
+
+Once you're on Playwright Test, you get a lot!
```
Comment (Meir017) [inline docs/src/testing-library-js.md:134]: consider mentioning the tests isolation

**Review by raghavthind2005 — APPROVED** [review]

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**